yetanotherco/aligned_layer
# feat (batcher): allow whitelist of non paying specific address

Batcher can be configured to allow a certain address to not pay for transactions.

Aligned cli uses hardcoded anvil address 9 instead of a random address.

# To test

Start everything as normal. Then you can test sending a proof with no keystore:
```
aligned submit \
--proving_system SP1 \
--proof ./batcher/aligned/test_files/sp1/sp1_fibonacci.proof \
--vm_program ./batcher/aligned/test_files/sp1/sp1_fibonacci-elf
```

It should go through. Then test sending a proof with a different keystore, it should fail because it has no funds:
```
aligned submit \
--proving_system SP1 \
--proof ./batcher/aligned/test_files/sp1/sp1_fibonacci.proof \
--vm_program ./batcher/aligned/test_files/sp1/sp1_fibonacci-elf \
--keystore_path ./config-files/anvil.batcher.ecdsa.key.json
```
